pyat/at/lattice/variable_elements.py

```diff
@@ -68,74 +89,103 @@ def __init__(self, family_name, AmplitudeA=None, AmplitudeB=None,
+            if modefromdict is not None:
+                mode = modefromdict  # matlab class
+            else:
+                mode = modepyatinput
+            self.Mode = int(mode)
+            # MaxOrder is later overwritten by lenth of amplitudes
+            self.MaxOrder = kwargs.get("MaxOrder", 0)
+            amplitudea = None
+            amplitudeb = None
+            if "AmplitudeA" in kwargs:
+                amplitudea = kwargs.pop("AmplitudeA", None)
+                amplitudea = self._set_params(amplitudea, mode, "A", **kwargs)
+            if "AmplitudeB" in kwargs:
+                amplitudeb = kwargs.pop("AmplitudeB", None)
+                amplitudeb = self._set_params(amplitudeb, mode, "B", **kwargs)
+            if amplitudea is not None:
+                self.AmplitudeA = amplitudea
+            if amplitudeb is not None:
+                self.AmplitudeB = amplitudeb
+            # end setting up Amplitudes and modes
+            kwargs.setdefault("PassMethod", "VariableThinMPolePass")
+            # this overwrites MaxOrder
+            self._setmaxorder(amplitudea, amplitudeb)
+            if mode == ACMode.WHITENOISE and "Seed" not in kwargs:
+                kwargs.update({"Seed": datetime.now().timestamp()})
+            self.Periodic = kwargs.pop("Periodic", True)
+            self.PolynomA = kwargs.pop("PolynomA", np.zeros(self.MaxOrder + 1))
+            self.PolynomB = kwargs.pop("PolynomB", np.zeros(self.MaxOrder + 1))
+            # check ramps
+            ramps = kwargs.pop("Ramps", None)
+            if ramps is not None:
+                if len(ramps) != 4:
+                    raise AtError("Ramps has to be a vector with 4 elements")
+                self.Ramps = ramps
+        # fill in super class
+        super().__init__(family_name, **kwargs)
+
+    def _setmaxorder(self, ampa: np.ndarray, ampb: np.ndarray):
         mxa, mxb = 0, 0
         if ampa is not None:
-            mxa = numpy.max(numpy.nonzero(ampa))
+            mxa = np.max(np.append(np.nonzero(ampa), 0))
         if ampb is not None:
-            mxb = numpy.max(numpy.nonzero(ampb))
+            mxb = np.max(np.append(np.nonzero(ampb), 0))
         self.MaxOrder = max(mxa, mxb)
         if ampa is not None:
             delta = self.MaxOrder - len(ampa)
             if delta > 0:
-                ampa = numpy.pad(ampa, (0, delta))
-            setattr(self, 'AmplitudeA', ampa)
+                ampa = np.pad(ampa, (0, delta))
+            self.AmplitudeA = ampa
         if ampb is not None:
             delta = self.MaxOrder + 1 - len(ampb)
             if delta > 0:
-                ampb = numpy.pad(ampb, (0, delta))
-            setattr(self, 'AmplitudeB', ampb)
+                ampb = np.pad(ampb, (0, delta))
+            self.AmplitudeB = ampb
 
-    def _set_params(self, amplitude, mode, ab, **kwargs):
+    def _set_params(
+        self, amplitude: int or str, mode, a_b: str, **kwargs: dict[str, any]
```

Comment:
No need to define the type of kwargs

Reply:
It is flake8 who suggested. Should I ignore it ? Do you work with some custom flake8 settings for pyat development ?